pkg/events/eventbus.go

```diff
@@ -0,0 +1,58 @@
+package events
+
+import (
+	"sync"
+
+	"github.com/pkg/errors"
+
+	"github.com/kumahq/kuma/pkg/core/resources/model"
+)
+
+func NewEventBus() *EventBus {
+	return &EventBus{}
+}
+
+type EventBus struct {
+	mtx         sync.RWMutex
+	subscribers []chan Event
+}
+
+func (b *EventBus) New() Reader {
+	b.mtx.Lock()
+	defer b.mtx.Unlock()
+
+	events := make(chan Event, 10)
```

Comment:
I'm just slightly concerned about this `10`. What will happen if one consumer of events won't process the events quickly enough? I think the sender will be stuck at `Send` operation right? What then happens in Postgres / K8S? Will we lose events or not?

Reply:
I looked at the implementation of Kubernetes/Postgress components that send events to EventBus. They have their own queues/channels and from what I saw they don't drop events. And right now we have only one consumer - `resyncer`. It uses `rateLimiter` and will drop events itself if will be too loaded.